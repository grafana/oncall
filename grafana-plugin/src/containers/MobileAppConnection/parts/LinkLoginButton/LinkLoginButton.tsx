import React, { FC } from 'react';

import { Button, Stack } from '@grafana/ui';

import { Text } from 'components/Text/Text';
import { StackSize } from 'utils/consts';

type Props = {
  baseUrl: string;
  token: string;
};

export const LinkLoginButton: FC<Props> = (props: Props) => {
  const { baseUrl, token } = props;
  const mobileDeepLink = `grafana://mobile/login/link-login?oncall_api_url=${baseUrl}&token=${token}`;

  return (
    <Stack direction="column" gap={StackSize.lg}>
      <Text type="primary" strong>
        Sign in via deeplink
      </Text>
      <Text type="primary">Make sure to have the app installed</Text>
      <Button
        variant="primary"
        onClick={() => {
          window.open(mobileDeepLink, '_blank');
        }}
      >
        Connect Mobile App
      </Button>
    </Stack>
  );
};
