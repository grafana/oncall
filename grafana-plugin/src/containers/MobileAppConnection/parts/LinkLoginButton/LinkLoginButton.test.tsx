import React from 'react';

import { render } from '@testing-library/react';

import { LinkLoginButton } from './LinkLoginButton';

describe('LinkLoginButton', () => {
  test('it renders properly', () => {
    const component = render(<LinkLoginButton baseUrl="http://test.url" token="test1213" />);
    expect(component.container).toMatchSnapshot();
  });
});
