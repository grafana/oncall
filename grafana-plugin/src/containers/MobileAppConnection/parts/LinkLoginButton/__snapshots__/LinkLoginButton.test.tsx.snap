// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkLoginButton it renders properly 1`] = `
<div>
  <div
    class="css-gjl87o-vertical-group"
    style="width: 100%; height: 100%;"
  >
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <span
        class="css-77ouhj--primary css-77ouhj--medium css-77ouhj--strong css-1287p17"
      >
        Sign in via deeplink
      </span>
    </div>
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <span
        class="css-77ouhj--primary css-77ouhj--medium css-1287p17"
      >
        Make sure to have the app installed
      </span>
    </div>
    <div
      class="css-12oo3x0-layoutChildrenWrapper"
    >
      <button
        class="css-td06pi-button"
        type="button"
      >
        <span
          class="css-1riaxdn"
        >
          Connect Mobile App
        </span>
      </button>
    </div>
  </div>
</div>
`;
